### ISSUE TITLE
Label Inference problem

### ISSUE THREAD

**Issue by CEclipse** [ashishrana160796/RerrFact#1]
Hi there,
I’ve followed your code steps but in label inference stage has some problem.
![fc64d1dbbb8647f9850a8a88233766a9](https://github.com/ashishrana160796/RerrFact/assets/107844121/c463adc0-a2ed-4eba-a6d7-be2eeaffd110)
Looking forward to your reply.

**Comment by ashishrana160796** [comment]
Hello @CEclipse 

From your error message it appears that you are doing a binary classification task in your problem. But, our entailment models were not fine-tuned for binary classification but rather three-way entailment problem. So, you either have to change your problem to a three-way learning and classification problem or follow any suitable approaches shared in these links _([link 1](https://stackoverflow.com/questions/69194640/mismatched-size-on-bertforsequenceclassification-from-transformers-and-multiclas) or [link 2](https://stackoverflow.com/questions/67158554/fine-tuning-models-classifier-layer-with-new-label/67187807#67187807))._

Please, feel free to close the issue if it resolves your problem. Thanks!

**Comment by CEclipse** [comment]
I've inference the  results by your label training and inference code. But you didnt pushlish your three-way entailment code. Could you tell me your whole pipeline code?

Looking forward to your reply.

**Comment by CEclipse** [comment]
But your label training code and label inference that is binary classification. I followed the steps it mentioned that mismatch. Could you please tell me the reason?

Looking forward to your reply

**Comment by ashishrana160796** [comment]
Hello @CEclipse!

I am sharing answers to each of your statements in pointer form, hope this format works out well.

* _I've inference the results by your label training and inference code. But you didnt pushlish your three-way entailment code. "_
  *  The label training and inference code is the "three-way entailment code" here.

*  _Could you tell me your whole pipeline code?_
	* Including training notebooks and inference scripts, the execution flow is _{abstract train -> abstract inference -> rationale train -> rationale inference -> label train -> label inference}_ where all these three modules have their separate fine-tuned models after the training execution. After each execution of inference the aim is to produce intermediate output files for further usage into the next module. The accompanying paper explains it on a functional level, [refer link](https://ceur-ws.org/Vol-3164/paper11.pdf).

* _But your label training code and label inference that is binary classification._
  * No, but I can see it can sound confusing if you are just starting out. Our label prediction methodology is not an easy one to understand, but it is quite effective for small datasets.

* _I followed the steps it mentioned that mismatch. Could you please tell me the reason?_
  * My best guess based on the information shared by you currently here is that the exact idea and code of the problem is not very clear to you as it is not well communicated by us. You are attempting a binary classification task according to the error message which is not implemented in the code base here.

Additionally, I am also sharing some other resources with you to help with your work here for the SCIFACT task, do look up their corresponding papers as well for more understanding.

* [SCIVER Baseline Approach](https://github.com/allenai/scifact)
* [QMUL SDS's Approach](https://github.com/XiaZeng0223/sciverbinary)

I hope this information was helpful for you. Please, feel free to close the issue if it resolves your problem. Also, it is quite the extensive pipeline and it might take you quite some time before it gets working completely. Thanks again!

**Comment by CEclipse** [comment]
Thank you for your patients. Thanks again!
